src/game_battlealgorithm.cpp

```diff
@@ -1117,8 +1117,10 @@ bool Game_BattleAlgorithm::Skill::Execute() {
 
 			if (skill.affect_hp)
 				this->hp = std::max<int>(0, std::min<int>(effect, GetTarget()->GetMaxHp() - GetTarget()->GetHp()));
-			if (skill.affect_sp)
-				this->sp = std::max<int>(0, std::min<int>(effect, GetTarget()->GetMaxSp() - GetTarget()->GetSp()));
+			if (skill.affect_sp) {
+				int prov_sp = GetSource() == GetTarget() ? source->CalculateSkillCost(skill.ID) : 0;
+				this->sp = std::max<int>(0, std::min<int>(effect, GetTarget()->GetMaxSp() - GetTarget()->GetSp() + prov_sp));
```

Comment:
This `prov_sp` is super ugly but its required here with the way we clamp early. This whole structure needs to be refactored. This wart is a reminder.

Reply:
what does "prov" even mean?

Reply:
No idea :)